### PR TITLE
feat: update failed test cases after release 3.1.2 for android

### PR DESCRIPTION
**Pull request by bilalawan321**
[LEARNER-8945](https://2u-internal.atlassian.net/browse/LEARNER-8945)

**Description**

Remove test cases for the discovery screen
update test cases for title of screens after updated ids
update strings file after version update to 3.1.2

### CODE REVIEW

File: tests/android/tests/test_android_my_courses_list.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def test_validate_ui_elements_smoke(self, set_capabilities, setup_logging):
 
         assert android_main_dashboard_page.load_courses_tab()
         # assert android_main_dashboard_page.get_profile_icon().text == strings.BLANK_FIELD
-        assert android_main_dashboard_page.get_title_textview().text == strings.MAIN_DASHBOARD_SCREEN_TITLE
+        assert android_main_dashboard_page.get_all_text_views()[0].text == strings.MAIN_DASHBOARD_SCREEN_TITLE
```
Comment (moeez96) [inline tests/android/tests/test_android_my_courses_list.py:44]: Possible index error?

File: tests/android/tests/test_android_course_html_component.py [hunk 2]
```diff
@@ -58,10 +58,8 @@ def test_ui_elements_smoke(self, set_capabilities, setup_logging):
 
         assert course_html_component.get_next_button().text == strings.COURSE_HTML_COMPONENT_NEXT_BUTTON
         assert course_html_component.get_prev_button().text == strings.COURSE_HTML_COMPONENT_PREV_BUTTON
-        assert course_html_component.get_next_unit_title()
```
Comment (moeez96) [inline tests/android/tests/test_android_course_html_component.py:61]: Please mention reason of removing asserts for next and prev unit titles.

File: tests/android/tests/test_android_course_discovery.py [hunk 3]
```diff
@@ -1,58 +0,0 @@
...
-"""
-
-from tests.android.pages.android_main_dashboard import AndroidMainDashboard
-from tests.android.pages.android_course_discovery import AndroidCourseDiscovery
-from tests.android.pages.android_login_smoke import AndroidLoginSmoke
-from tests.common import strings
-from tests.common.globals import Globals
-
-
-class TestAndroidCourseDiscovery(AndroidLoginSmoke):
```
[the remark this answers is not in the document]
Comment (bilalawan321) [reply tests/android/tests/test_android_course_discovery.py:12]: New discovery experience is shifted to Webview now in this release 3.1.2, so we need to remove the previous native test cases.

File: tests/android/tests/test_android_my_courses_list.py (hunk 1 above)
Comment (bilalawan321) [reply tests/android/tests/test_android_my_courses_list.py:44]: the test case is expecting the title of the screen on this specific index. if the title is not on this index then the test case will. be failed, which is the expected behavior.

**Pull request merged**